### PR TITLE
feat: federate SampleForReport object

### PR DESCRIPTION
**Pull request by ehoops-cz**
# Pull Request

## JIRA Ticket

[CZID-9440](https://czi-tech.atlassian.net/browse/CZID-9440)

## Description

This adds federation for the SampleForReport query. 

There is logic here to pull from Rails or to acknowledge the feature flag and pull from next gen. 

### To Do In Next PR
  - deprecated - this is not where we expect it to be in the sandbox data - I will investigate and fix in a different PR
  - add clause to workflowQuery to specify "consensus-genome" rather than "bulk-download"
  - more advanced error handling if one of the queries do not work

## Tests

- Unit tests added
- Smoke Tests on the front end included testing the UI in sandbox and on local for functionality with and without the feature flag on. 

[CZID-9440]: https://czi-tech.atlassian.net/browse/CZID-9440?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .meshrc.yaml [hunk 1]
```diff
@@ -96,6 +96,18 @@ sources:
             requestSample: { pipelineVersion: "8.0" }
             responseSchema: ./json-schemas/sampleMetadata.json
             responseTypeName: SampleMetadata
+          - type: Query
+            field: SampleForReport
+            method: GET
+            path: /samples/{args.railsSampleId}.json
+            responseSchema: ./json-schemas/sampleForReportResponse.json
+            responseTypeName: SampleForReport
+          - type: Query
```
Comment (bchu1) [inline .meshrc.yaml:105]: Why are there 2?

File: json-schemas/sampleForReportRequest.json
Comment (bchu1) [inline json-schemas/sampleForReportRequest.json:1]: I don't think this is valid JSON schema (need the `"type"` thing)

File: resolvers.ts [hunk 2]
```diff
@@ -455,6 +460,211 @@ export const resolvers: Resolvers = {
...
+      const nextGenSampleId = entitiesResp?.data.samples[0].id;
+      const workflowsQuery = `
+          query WorkflowsQuery {
+            workflowRuns(where: {entityInputs: {inputEntityId: {_eq: "${nextGenSampleId}"}}}) {
+              id
+              _id
+              railsWorkflowRunId
+              status
+              ownerUserId
+              deprecated
```
Comment (bchu1) [inline resolvers.ts:594]: I think this is now `deprecatedById` or something

File: resolvers.ts [hunk 3]
```diff
@@ -455,6 +460,211 @@ export const resolvers: Resolvers = {
...
+      // ✅ 🤷‍♀️ input_error
+      // 🤷‍♀️ run_finalized
+      // ✅ 🤷‍♀️ wdl_version
+
+      // 2. ⏰ parse ref_fasta from entitiesResp
+      // 3. ✅ 🤷‍♀️ specify "consensus-genome" workflow type from workflowsQuery
+      // 4. 🤷‍♀️ add clause to workflowQuery to specify "consensus-genome" rather than "bulk-download"
+      // 5. ✅ parse creationSource from rawInputsJson
+
+      const entitiesQuery = `
```
Comment (bchu1) [inline resolvers.ts:533]: This craziness makes sense to me. Sad that we have to do it but I don't see any other way 😢

File: .meshrc.yaml (hunk 1 above)
Comment (rainandbare) [reply .meshrc.yaml:105]: There shouldn't be!

**Review by bchu1 — APPROVED** [review]

**Pull request merged**